hack/generator/pkg/astmodel/type_name_processing_register.go

```diff
@@ -0,0 +1,100 @@
+package astmodel
+
+// TypeNameProcessingRegister is a register of TypeNames that require processing, keeping track of which names have been
+// processed and which are still outstanding. This is used, for example, for a consistency check that all TypeName
+// references can be satisfied
+type TypeNameProcessingRegister struct {
```

Comment:
Do you have an example of how this is used? There are two things that I'm struggling with:

1. Does this belong in `astmodel` or is it actually specific to the storage processing? Trying to understand how it might be used outside of the storage type generation context, but first it would probably help to understand how it's being used in that context.
2. As discussed in person, it's not clear to me if `TypeWalker` can help you do this or not. Obviously the two things are very different - seeing how this was used (maybe in a draft PR?) might help me answer this question.